guides/rails_guides/kindle.rb

```diff
@@ -17,7 +17,7 @@ def generate(output_dir, mobi_outfile, logfile)
       puts "=> Arranging html pages in document order"
       toc = File.read("toc.ncx")
       doc = Nokogiri::XML(toc).xpath("//ncx:content", "ncx" => "http://www.daisy.org/z3986/2005/ncx/")
-      html_pages = doc.select { |c| c[:src] }.map { |c| c[:src] }.uniq
+      html_pages = doc.filter_map { |c| c[:src] }.uniq
```

Comment:
While this one is clearly good, the rest of these seem like aesthetic / comprehension downgrades to me, and none are in places the performance is interesting.